src/main/java/org/fundacionjala/coding/wilson/EanValidation.java

```diff
@@ -0,0 +1,39 @@
+package org.fundacionjala.coding.wilson;
+
+/**
+ *
+ */
+public final class EanValidation {
+
+    /**
+     *
+     */
+    private EanValidation() {
+    }
+
+    /**
+     *
+     * @param eanCode is a cod
+     * @return suma control
+     */
+    public static boolean validate(final String eanCode) {
+        final char base = '0';
+        final int limited = 12;
+        final int base2 = 10;
+
+        final int impar = 1;
+        final int par = 3;
+
+        int sum = 0;
+        for (int i = 0; i < limited; i++) {
+            int c = eanCode.charAt(i) - base;
+            sum += c % 2 != 0 ? c * impar : c * par;
+        }
+        if (sum % base2 == 0) {
+            return eanCode.charAt(limited) == base;
+        } else {
+            sum = base2 - (sum % base2);
+            return eanCode.charAt(limited) - base == sum;
+        }
```

Comment:
This may be refactored like this.
```java
        int checkSum = sum % base2 == 0 ? 0 : base2 - (sum % base2);
        return Character.getNumericValue(eanCode.charAt(limited)) == checkSum;
```